### PR TITLE
Refactor ExecutionTests into a separate dll

### PR DESCRIPTION
**Pull request by hekota**
ExecutionTests are currently compiled into clang-hlsl-tests.dll together with all of the other DXC unit tests. Later in the code pipeline the ExecutionTests are getting transformed into HLK tests which have different dependency constraints than the clang HLSL tests. 

Refactoring ExecutionTests into a separate dll allows us to set its dependencies and build settings accordingly and closely matching those of the HLK test dll. This will help HLK authors to implement the tests with those constraints in mind.

### CODE REVIEW

File: include/dxc/Test/HlslTestUtils.h [hunk 1]
```diff
@@ -336,8 +371,9 @@ inline bool GetTestParamBool(LPCWSTR name) {
   if (NameValue.IsEmpty()) {
     return false;
   }
-  return Unicode::IsStarMatchWide(ParamValue, ParamValue.GetLength(),
-                                  NameValue, NameValue.GetLength());
+
+  return hlsl_test::IsStarMatchWide(ParamValue, ParamValue.GetLength(),
```
Comment (python3kgae) [inline include/dxc/Test/HlslTestUtils.h:375]: Why cannot use Unicode::IsStarMatchWide here?

Comment (python3kgae) [reply include/dxc/Test/HlslTestUtils.h:375]: Because we don't want to depend on dxcSupport?

File: include/dxc/Test/HlslTestUtils.h [hunk 2]
```diff
@@ -29,7 +29,8 @@
 #include "dxc/DXIL/DxilConstants.h" // DenormMode
```
Comment (python3kgae) [inline include/dxc/Test/HlslTestUtils.h:29]: Remove include Unicode.h?

File: include/dxc/Test/HlslTestUtils.h (hunk 1 above)
Comment (hekota) [reply include/dxc/Test/HlslTestUtils.h:375]: Yes, we want to avoid linking DXC support lib to the ExecutionTests and therefore the HLK tests.

**Review by python3kgae — APPROVED** [review]

**Pull request merged**